account_invoice_production_lot/models/account_invoice.py

```diff
@@ -34,7 +34,9 @@ class AccountInvoiceLine(models.Model):
     @api.multi
     def _compute_prod_lots(self):
         for line in self:
-            line.prod_lot_ids = line.move_line_ids.mapped('lot_ids')
+            quant_ids = line.move_line_ids.mapped('quant_ids').filtered(
+                lambda x: x.location_id.usage == 'customer')
```

Comment:
@tafaRU why didn't you keep the other filters on the moves too?
```
not move.origin_returned_move_id or (move.origin_returned_move_id and move.to_refund_so)
```
source (from the same method linked earlier): https://github.com/odoo/odoo/blob/0b414cb8145381cd8a3946aae9cec1601b111238/addons/sale_stock/models/sale_order.py#L185

Reply:
@tafaRU I have created https://github.com/tafaRU/account-invoice-reporting/pull/1.
The filters that I suggested in https://github.com/OCA/account-invoice-reporting/pull/99#discussion_r255967022 aren't completely feasible with this module's purpose because, for what I read from the README and the tests, it doesn't care about refunds (field `to_refund_so` in the return wizard) or the `quantity` returned.
That's why I only took a few lines from the core implementations, and tried to manage returned moves.
I added comments too, hope it's clear enough.

Reply:
> - refunds: field to_refund_so in the return wizard is never mentioned
> - the quantity returned: if a lot is returned, it counts as returned regardless it was completely returned or not


@SimoRubi the implementation based on quants available in customer location should not be affected by this problems

Reply:
@eLBati in https://github.com/tafaRU/account-invoice-reporting/pull/1 I tried to implement a solution to what was concerning me in https://github.com/OCA/account-invoice-reporting/pull/99#discussion_r255967022: I tried to mimic the behavior of Odoo core in order to have consistent behavior.

If Odoo core counts the delivered qty [this way](https://github.com/odoo/odoo/blob/0b414cb8145381cd8a3946aae9cec1601b111238/addons/sale_stock/models/sale_order.py#L176-L189) maybe we should do so too and there are some implications that we don't know in looking only to quants.

Reply:
Here we need the delivered lots, not the quantities